docs/docs/guides/build/assets/asset-selection-syntax/reference.md

```diff
@@ -0,0 +1,100 @@
+| **Two layers upstream of an asset** | `2+key:"my_asset"` | Selects assets two steps upstream from `my_asset`. |
+| **All layers downstream of an asset** | `key:"my_asset"+` | Selects all downstream assets that depend on `my_asset`. |
+| **One layer downstream of an asset** | `key:"my_asset"+1` | Selects assets that directly receive input from `my_asset`. |
+| **Two layers downstream of an asset** | `key:"my_asset"+2` | Selects assets two steps downstream from `my_asset`. |
+| **One layer upstream and downstream of an asset** | `1+key:"my_asset"+1` | Selects one layer of assets providing input to and receiving input from `my_asset`. |
+| **Two layers upstream and downstream of an asset** | `2+key:"my_asset"+2` | Selects two layers of assets providing input to and receiving input from `my_asset`. |
+| **All layers upstream and downstream of an asset** | `+key:"my_asset"+` | Selects all assets upstream and downstream of `my_asset`. |
+
+## Operands and grouping
+
+You can combine multiple filters with operands and group them with parentheses to further refine your asset selection:
+
+| Operand | Syntax | Description |
+|---------|--------|-------------|
+| **`and`** | `owner:"alice" and kind:"dbt"` | Selects assets owned by `alice` of kind `dbt`. |
+| **`or`** | `owner:"billing" or owner:"sales"` | Selects assets owned by either `billing` or `sales`. |
+| **`not`** | `not tag:"obsolete"` | Excludes assets tagged with `obsolete`. |
+| **Grouping `()`** | `(owner:"alice" or group:"analytics") and  kind:"table"` | Symbols `(` and `)` used to group expressions and control the order of evaluation in queries. This example selects assets that are both owned by `alice` and of kind `table`, or that belong to the `analytics` group. |
+
+## Functions
+
+You can use `roots(selection)` and `sinks(selection)` functions to return the sink or root assets of an asset selection.
+
+- **Root assets** are assets that have no upstream dependencies within the asset selection. Root assets can have upstream dependencies outside of the asset selection.
+
+- **Sink assets** are assets that have no downstream dependencies within the asset selection. Sink assets can have downstream dependencies outside of the asset selection.
+
+| Example | Description |
+|--------|--------------|
+| `roots(*)` | Select all assets without any upstream dependencies. |
+| `sinks(*)` | Select all assets without any downstream dependencies. |
+| `roots(group:"public_data")` | Selects root assets within the `public_data` group.|
+| `sinks(group:"public_data")` | Selects sink assets within the `public_data` group. |
+| `roots(+group:"public_data")` | Selects all root assets that feed into the `public_data` group, but do not belong to that group. |
+| `sinks(group:"public_data"+)` | Selects all sink assets that depend on assets in the `public_data` group, but do not belong to that group. |
```

Comment:
```suggestion
| `roots(+group:"public_data")` | Selects all root assets that feed into the `public_data` group. |
| `sinks(group:"public_data"+)` | Selects all sink assets that depend on assets in the `public_data` group. |
```

Reply:
The sink/roots could belong to the group.

Reply:
I guess maybe the intent is to signal that they could potentially be outside of the group? We could say something like 

"Selects all sink assets that depend on assets in the `public_data` group, they may or may not belong to the `public_data` group themselves."